src/Features/CSharp/Portable/BraceCompletion/AbstractCurlyBraceOrBracketCompletionService.cs

```diff
@@ -115,18 +114,24 @@ private static bool ContainsOnlyWhitespace(SourceText text, int openingPosition,
             if (closingPointLine - openingPointLine == 1)
             {
                 // Handling syntax tree directly to avoid parsing in potentially UI blocking code-path
-                var closingToken = document.Root.FindTokenOnLeftOfPosition(context.ClosingPoint);
-                var newLineString = options.FormattingOptions.NewLine;
-                newLineEdit = new TextChange(new TextSpan(closingToken.FullSpan.Start, 0), newLineString);
+                var closingToken = document.Root.FindToken(closingPoint - 1);
+                Debug.Assert(IsValidClosingBraceToken(closingToken));
```

Comment:
i dont' love that thsi is only an assert.  maybe bail out here since if we don't find the close token, everything is going to go bad?

Reply:
I don't think these are actually needed. If we can't find the token here then something outside this method must be seriously wrong. In that case I guess we'd like this to crash so we can get dump?

Reply:
i guess my question is: how do we know we'll have a close token here?  I don't know this code well enough, so this is a question to you about what code ran prior to this that absolutely made us certain we have a good tree, and the tree will think there's a clsoe brace at this position :)

Reply:
Ah, I see. From what I saw, the code prior handles the insertion of closing brace [here](https://sourceroslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/AutomaticCompletion/BraceCompletionSessionProvider.BraceCompletionSession.cs,119). Only then this method would be called to handle 'return'